packages/block-editor-tools/src/hooks/use-post/index.js

```diff
@@ -9,7 +9,7 @@ import { useSelect } from '@wordpress/data';
  *                   and the returned post object.
  */
 const usePost = (postId, postType = 'post') => useSelect(
-  (select) => select('core').getEntityRecord('postType', postType, postId),
+  (select) => select('core').getEntityRecord('postType', postType, postId, { context: 'view' }),
```

Comment:
Cool, this is a great solution for the first part of the AC. To accomplish the other part of the AC, which makes it possible to fetch it in an edit context, can you extend both `usePost` and `usePostById` to include an optional third parameter that includes this as the default? Basically:

```js
const usePost = (postId, postType = 'post', options = { context: 'view' }) => // etc
```

And do that for both usePost and usePostById, and have usePostById pass its options parameter to this function.

Reply:
That should be doable, for sure.